### PR TITLE
Support PublishAot in addition to RunAOTCompilation for doing Mono-based AOT

### PR DESCRIPTION
**Pull request by akoeplinger**
Sibling PRs:
- https://github.com/dotnet/sdk/pull/34051
- https://github.com/dotnet/emsdk/pull/375

### CODE REVIEW

File: src/mono/wasm/build/WasmApp.Native.targets [hunk 1]
```diff
@@ -104,13 +104,13 @@
     <Error Condition="'$(WasmBuildNative)' == 'true' and '$(_IsEMSDKMissing)' == 'true'"
            Text="$(_EMSDKMissingErrorMessage) Emscripten SDK is required for building native files." />
 
-    <Error Condition="'$(RunAOTCompilation)' == 'true' and '$(_IsEMSDKMissing)' == 'true'"
+    <Error Condition="('$(RunAOTCompilation)' == 'true' or '$(PublishAot)' == 'true') and '$(_IsEMSDKMissing)' == 'true'"
```
[the remark this answers is not in the document]
Comment (steveisok) [reply src/mono/wasm/build/WasmApp.Native.targets:107]: If `PublishAotUsingMonoRuntime` was != true, that would mean nativeaot in the platforms that support it.

**Comment by radical** [comment]
I wonder what we should do in the tests which use only `RunAOTCompilation`. Randomly pick one or the other property? Run some with `RunAOTCompilation`, and the rest with the new one? Or a better idea?
cc @lewing

File: src/mono/wasi/build/WasiApp.Native.targets [hunk 2]
```diff
@@ -60,7 +60,7 @@
     <!-- When Building -->
     <PropertyGroup Condition="'$(WasmBuildingForNestedPublish)' != 'true'">
       <!-- build AOT, only if explicitly requested -->
-      <WasmBuildNative Condition="'$(RunAOTCompilation)' == 'true' and '$(RunAOTCompilationAfterBuild)' == 'true'">true</WasmBuildNative>
```
Comment (vargaz) [inline src/mono/wasi/build/WasiApp.Native.targets:63]: Wouldn't it be simpler to set RunAOTCompilation if PublishAot is set ?

Comment (akoeplinger) [reply src/mono/wasi/build/WasiApp.Native.targets:63]: We could, but the idea is that we can remove RunAOTCompilation once we no longer support releases that used it.

**Comment by akoeplinger** [comment]
> I wonder what we should do in the tests which use only `RunAOTCompilation`. Randomly pick one or the other property? Run some with `RunAOTCompilation`, and the rest with the new one? Or a better idea?

My idea was to remove all usages of RunAOTCompilation from our tests once we're using a new enough SDK that supports the new property.

**Comment by akoeplinger** [comment]
Closing for now until we have a decision on whether we want to do this.

**Pull request closed without merging**